snappy/overlord.go

```diff
@@ -175,36 +175,26 @@ func UndoCopyData(newSnap *Snap, flags InstallFlags, meter progress.Meter) {
 	}
 }
 
-func ActivateSnap(s *Snap, inter interacter) error {
-	currentActiveSymlink := filepath.Join(s.basedir, "..", "current")
-	currentActiveDir, _ := filepath.EvalSymlinks(currentActiveSymlink)
-
-	// already active, nothing to do
-	if s.basedir == currentActiveDir {
-		return nil
-	}
+func GenerateSecurityProfile(s *Snap) error {
+	// generate the security policy from the snap.yaml
+	// Note that this must happen before binaries/services are
+	// generated because serices may get started
+	instDir := filepath.Join(dirs.SnapSnapsDir, s.Name(), s.Version())
+	return generatePolicy(s.m, instDir)
```

Comment:
Why do we have (almost) two local names for the same function?  Can't we simply move the content of generatePolicy here and use that instead?